### PR TITLE
ports/esp32: On md5 hashlib.

### PR DESCRIPTION
**Pull request by IhorNehrutsa**

### CODE REVIEW

File: extmod/modhashlib.c [hunk 1]
```diff
@@ -297,7 +297,7 @@ STATIC mp_obj_t hashlib_md5_digest(mp_obj_t self_in) {
 
 #if MICROPY_SSL_MBEDTLS
 
-#if MBEDTLS_VERSION_NUMBER < 0x02070000
+#if MBEDTLS_VERSION_NUMBER < 0x02070000 || MBEDTLS_VERSION_NUMBER >= 0x03000000
```
Comment (dpgeorge) [inline extmod/modhashlib.c:300]: Did mbedtls rename these functions again in v3?

Comment (IhorNehrutsa) [reply extmod/modhashlib.c:300]: https://github.com/micropython/micropython/blob/f6d630877c3b31a5091dff2fd615ddb92189a46c/extmod/modhashlib.c#L78-L82

https://github.com/micropython/micropython/blob/f6d630877c3b31a5091dff2fd615ddb92189a46c/extmod/modhashlib.c#L206-L210

Comment (dpgeorge) [reply extmod/modhashlib.c:300]: Ah, I see.

**Comment by dpgeorge** [comment]
Merged, thanks!

**Pull request merged**